src/main/java/org/veupathdb/service/eda/ss/model/db/VariableFactory.java

```diff
@@ -74,15 +76,19 @@ static Variable createVariableFromResultSet(ResultSet rs, Entity entity) throws
         getRsOptionalString(rs, VARIABLE_PARENT_ID_COL_NAME, null),
         getRsOptionalString(rs, DEFINITION_COL_NAME, ""),
         parseJsonArrayOfString(rs, HIDE_FROM_COL_NAME));
-
+    // Only set binary properties if binaryMetadataProvider is present.
```

Comment:
Rather than using null for "not present" for the BinaryMetadataProvider, maybe use an Optional as the passed arg and the value of the field.  Another option is to subclass for the case that needs it (where it lives as a field), and whenever it is used, add no-op overrideable methods that use the subclass's field when needed.

Reply:
Fixed this by having multiple implementations of `VariableFactory`